api/src/filters/exception.filter.ts

```
@@ -77,8 +101,25 @@ export class CustomExceptionFilter implements ExceptionFilter {
   private isUniqueConstraintViolation(exception: any): boolean {
     return (
       exception.status === 409 ||
-      (exception.name === 'QueryFailedError' && exception.code === 'ER_DUP_ENTRY') ||
-      exception.code === 'SQLITE_CONSTRAINT_UNIQUE'
+      (exception.name === 'QueryFailedError' && exception.code === 'ER_DUP_ENTRY') || // MySQL
+      (exception.name === 'QueryFailedError' && exception.code === '23505') || // PostgreSQL unique constraint violation
+      (exception.name === 'SqliteError' && exception.code === 'SQLITE_CONSTRAINT_UNIQUE') || // SQLite unique constraint
+      (exception.name === 'QueryFailedError' && exception.code === 'SQLITE_CONSTRAINT_UNIQUE') || // SQLite via TypeORM
+      (exception.errno === 19 && exception.code === 'SQLITE_CONSTRAINT') || // SQLite constraint error
+      (exception.message && exception.message.includes('UNIQUE constraint failed')) // SQLite unique constraint message
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)